Packet++/src/LdapLayer.cpp

```diff
@@ -214,6 +330,77 @@ namespace pcpp {
 	}
 	// endregion
 
+	// region LdapResponseLayer
+
+	LdapResponseLayer::LdapResponseLayer(uint16_t messageId, const LdapOperationType& operationType, const LdapResultCode& resultCode,
+		const std::string& matchedDN, const std::string& diagnosticMessage, const std::vector<std::string>& referral,
+		const std::vector<LdapControl>& controls)
+	{
+		Asn1EnumeratedRecord resultCodeRecord(resultCode);
+		Asn1OctetStringRecord matchedDNRecord(matchedDN);
+		Asn1OctetStringRecord diagnosticMessageRecord(diagnosticMessage);
+
+		std::vector<Asn1Record*> messageRecords = {&resultCodeRecord, &matchedDNRecord, &diagnosticMessageRecord};
+
+		std::unique_ptr<Asn1ConstructedRecord> referralRecord;
```

Comment:
nit: Does `referralRecord` need to exist outside the `if` scope?

Reply:
Nvm, I see you are using the pointer for memory management and the raw pointer is used in `init`